src/ScottPlot/Plottable/RadialGaugePlot.cs

```diff
@@ -0,0 +1,709 @@
+        /// Size of the text is set by <see cref="GaugeLabelsFontPct"/> and color by <see cref="GaugeLabelsColor"/>.
+        /// </summary>
+        public bool ShowGaugeValues { get; set; } = true;
+
+        public System.Drawing.Drawing2D.LineCap EndCap { get; set; } = System.Drawing.Drawing2D.LineCap.Triangle;
+
+        public System.Drawing.Drawing2D.LineCap StartCap { get; set; } = System.Drawing.Drawing2D.LineCap.Round;
+
+        /// <summary>
+        /// Controls rendering style of the concentric circles (ticks) of the web
+        /// </summary>
+        //public RadarAxis AxisType { get; set; } = RadarAxis.None;
+
+        // These 3 properties are needed as part of IPlottable
+        public bool IsVisible { get; set; } = true;
+        public int XAxisIndex { get; set; } = 0;
+        public int YAxisIndex { get; set; } = 0;
+
+        #endregion Properties & fields
+
+        /// <summary>
+        /// Initializes the plot instance.
+        /// </summary>
+        /// <param name="values">Array of values to be plotted as gauges.</param>
+        /// <param name="lineColors">Array colors for the gauges.</param>
+        public RadialGaugePlot(double[] values, Color[] lineColors)
+        {
+            GaugeColors = lineColors;
+            Update(values);
+        }
+
+        public override string ToString() =>
+            $"RadialGauge with {DataRaw.Length} points.";
+
+        /// <summary>
+        /// Replace the data values with new ones. This passed data is copied and stored in <see cref="DataRaw"/>.
+        /// Implicitly calls the <see cref="ComputeAngularData"/> routine
+        /// </summary>
+        /// <param name="values">Array of values to be plotted as gauges.</param>
+        public void Update(double[] values)
+        {
+            DataRaw = new double[values.Length];
+            Array.Copy(values, 0, DataRaw, 0, values.Length);
+
+            // Compute MaxScale and MinScale values and transform DataRaw into DataAngular
+            ComputeMaxMin();
+            ComputeAngularData();
+        }
+
+        /// <summary>
+        /// Returns the original data in case it should be needed.
+        /// </summary>
+        /// <returns>Original data vector</returns>
+        public double[] GetData() => DataRaw;
+
+        /// <summary>
+        /// Returns the computed angular data in case it should be needed.
+        /// </summary>
+        /// <returns>Angular data vector</returns>
+        public double[,] GetAngularData() => DataAngular;
+
+        /// <summary>
+        /// Converts <see cref="DataRaw"/> into <see cref="DataAngular"/>.
+        /// Depends on <see cref="DataRaw"/>, <see cref="GaugeMode"/>, <see cref="GaugeDirection"/>, <see cref="StartingAngleGauges"/>, <see cref="AngleRange"/>, and <see cref="MaxScale"/>
+        /// </summary>
+        private void ComputeAngularData()
+        {
+            // Check if there's data
+            if (DataRaw == null) return;
```

Comment:
I think this should throw, not fail silently.

And maybe the null check should be in update?

Reply:
I don't know what would be the best options here.

Reply:
I'm refactoring these setters to throw exceptions rather than trim the data into compliance